protocol/communities/community.go

```diff
@@ -273,6 +281,24 @@ func (o *Community) DescriptionText() string {
 	return ""
 }
 
+func (o *Community) IntroMessage() string {
+	if o != nil &&
```

Comment:
I've followed the current file convention, assumed these checks are needed in some cases, see:
```
func (o *Community) Name() string {
	if o != nil &&
		o.config != nil &&
		o.config.CommunityDescription != nil &&
		o.config.CommunityDescription.Identity != nil {
		return o.config.CommunityDescription.Identity.DisplayName
	}
	return ""
}
func (o *Community) DescriptionText() string {
	if o != nil &&
		o.config != nil &&
		o.config.CommunityDescription != nil &&
		o.config.CommunityDescription.Identity != nil {
		return o.config.CommunityDescription.Identity.Description
	}
	return ""
}
func (o *Community) MembersCount() int {
	if o != nil &&
		o.config != nil &&
		o.config.CommunityDescription != nil {
		return len(o.config.CommunityDescription.Members)
	}
	return 0
}
```

Reply:
In golang you can call a function on a nil point :exploding_head: 

You can check this playground https://go.dev/play/p/x2xJIKsG2Dj , basically you can call functions on a `nil` object as long as they don't refer to themselves, so the checks above make some sense. 
In theory, the caller should always check for nils, so they should never be called, and having those checks might actually not be beneficial since you might just be hiding or delaying the issue (though we don't want to panic as that crashes the whole app, we should return an error), but that's contextual, can't remember/don't know why we do it that way, but seems like if the community is nil, we should just initiate an empty struct, so those checks looks a bit pesky.

Reply:
>  but that's contextual, can't remember/don't know why we do it that way  

So I will leave it as is, for now, don't want to risk any regression.